File: INTRODUCCION A POO/EJERCICIO-1/Persona.java
public class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public String mostrarSaludo() {
        return "Hola, soy " + this.nombre + " de " + this.ciudad;
    }

    public boolean esMayorDeEdad() {
        return this.edad >= 18;
    }

    public static void main(String[] args) {
        Persona persona1 = new Persona("Ana", 25, "Madrid");
        Persona persona2 = new Persona("Luis", 16, "Barcelona");
        Persona persona3 = new Persona("Sofía", 30, "Valencia");

        System.out.println(persona1.mostrarSaludo());
        System.out.println(persona2.mostrarSaludo());
        System.out.println(persona3.mostrarSaludo());
    }
}

File: INTRODUCCION A POO/EJERCICIO-1/Persona.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad
    
    def mostrar_saludo(self):
        return f"Hola, soy {self.nombre} de {self.ciudad}"
    
    def es_mayor_de_edad(self):
        return self.edad >= 18

if __name__ == "__main__":
    persona1 = Persona("Ana", 25, "Madrid")
    persona2 = Persona("Luis", 16, "Barcelona")
    persona3 = Persona("Sofía", 30, "Valencia")
    
    print(persona1.mostrar_saludo())
    print(persona2.mostrar_saludo())
    print(persona3.mostrar_saludo())
    